src/Compilers/CSharp/Test/Semantic/Semantics/UsingStatementTests.cs

```diff
@@ -476,243 +345,125 @@ static void Main()
         public void UsingPatternExtensionMethodPropertyHidingTest()
         {
             var source = @"
-class C1
+ref struct S1
 {
-    public C1() { }
-
     public int Dispose { get; }
 }
 
 static class C2 
 {
-    public static void Dispose(this C1 c1) { }
-}
-
-class C3
-{
-    static void Main()
-    {
-        using (C1 c = new C1())
-        {
-        }
-        C1 c1b = new C1();
-        using (c1b) { }
-    }
-}";
-            CreateCompilation(source).VerifyDiagnostics();
-        }
-
-        [Fact]
-        public void UsingPatternExtensionMethodInaccessibleInstanceMethodTest()
-        {
-            var source = @"
-using System;
-class C1
-{
-    public C1() { }
-
-    protected void Dispose() { }
-}
-
-static class C2 
-{
-    public static void Dispose(this C1 c1)
-    {
-        Console.WriteLine(""C2.Dispose(C1)"");
-    }
+    public static void Dispose(this S1 s1) { }
 }
 
 class C3
 {
     static void Main()
     {
-        using (C1 c = new C1())
+        using (S1 s = new S1())
         {
         }
+        S1 s1b = new S1();
+        using (s1b) { }
     }
 }";
             CreateCompilation(source).VerifyDiagnostics();
-
-            CompileAndVerify(source, expectedOutput: "C2.Dispose(C1)");
         }
 
         [Fact]
         public void UsingPatternStaticMethodTest()
         {
             var source = @"
-class C1
+ref struct S1
 {
-    public C1() { }
-
     public static void Dispose() { }
 }
 
 class C2
 {
     static void Main()
     {
-        using (C1 c = new C1())
-        {
-        }
-    }
-}";
-            CreateCompilation(source).VerifyDiagnostics(
-                // (14,16): error CS0176: Member 'C1.Dispose()' cannot be accessed with an instance reference; qualify it with a type name instead
-                //         using (C1 c = new C1())
-                Diagnostic(ErrorCode.ERR_ObjectProhibited, "C1 c = new C1()").WithArguments("C1.Dispose()").WithLocation(13, 16),
-                // (14,16): error CS1674: 'C1': type used in a using statement must be implicitly convertible to 'System.IDisposable' or have a public void-returning Dispose() instance method.
-                //         using (C1 c = new C1())
-                Diagnostic(ErrorCode.ERR_NoConvToIDisp, "C1 c = new C1()").WithArguments("C1").WithLocation(13, 16)
-                );
-        }
-
-
-        [Fact]
-        public void UsingPatternHidingInvalidInheritedWithPropertyAndValidExtensionMethodTest()
```

Comment:
>UsingPatternHidingInvalidInheritedWithPropertyAndValidExtensionMethodTest [](start = 20, length = 73)

Are we testing an invalid method on `ref struct` and valid extension method?

Reply:
Added